diego_mendizabal/tic_tac_toe/test/MainTest.java

```diff
@@ -0,0 +1,214 @@
+    }
+
+    /**
+     * Verifies if the player can place a move with symbol X
+     */
+    @Test
+    void playerMoveX() {
+        char [][] board = main.generateBoard();
+        main.playerMove('x',1, board);
+        assertEquals(board[0][0], 'x', "values does not match verify insertion position");
+    }
+
+    /**
+     * Verifies if the player can place a move with symbol O
+     */
+    @Test
+    void playerMove0() {
+        char [][] board = main.generateBoard();
+        main.playerMove('0',9, board);
+        assertEquals(board[4][4], '0', "values does not match verify insertion position");
+    }
+
+    /**
+     * Verifies if the board is full
+     */
+    @Test
+    void verifyBoardIsFull() {
+        char [][] tablero = main.generateBoard();
+        tablero[0][0] = 'X';// pos 1
+        tablero[0][2] = 'O';// pos 2
+        tablero[0][4] = 'X';// pos 3
+        tablero[2][0] = 'O';// pos 4
+        tablero[2][2] = 'X';// pos 5
+        tablero[2][4] = 'O';// pos 6
+        tablero[4][0] = 'O';// pos 7
+        tablero[4][2] = 'X';// pos 8
+        tablero[4][4] = 'O';// pos 9
+        assertTrue(main.verifyBoardIsFull(tablero), "board verification fails review board data");
+    }
+
+    /**
+     * Verifies if the board is not full
+     */
+    @Test
+    void verifyBoardIsNotFull() {
+        char [][] tablero = main.generateBoard();
+        tablero[0][0] = 'X';// pos 1
+        tablero[0][2] = 'O';// pos 2
+        tablero[0][4] = 'X';// pos 3
+        tablero[2][0] = 'O';// pos 4
+        tablero[2][2] = ' ';// pos 5
+        tablero[2][4] = 'O';// pos 6
+        tablero[4][0] = 'O';// pos 7
+        tablero[4][2] = ' ';// pos 8
+        tablero[4][4] = 'O';// pos 9
+        assertFalse(main.verifyBoardIsFull(tablero), "board verification fails review board data");
+    }
+
+    /**
+     * Verifies insert a character at certain position in the board
+     */
+    @Test
+    void insertCharacterAtPosition() {
+        char [][] board = main.generateBoard();
+        main.insertCharacterAtPosition('X', board,2,2);
+        assertEquals(board[2][2], 'X', "values was not inserted");
+    }
+
+    /**
+     * evaluates if the game is over at win for Diagonal line
+     */
+    @Test
+    void evaluatorDiagonal() {
+        char [][] board = main.generateBoard();
+        main.playerMove('0',1, board);
+        main.playerMove('0',5, board);
+        main.playerMove('0',9, board);
+        assertTrue(main.evaluator(board), "review evaluator at positions 1,5,9");
+    }
+
+    /**
+     * evaluates if the game is over at win for Column line
+     */
+    @Test
+    void evaluatorColumn() {
+        char [][] board = main.generateBoard();
+        main.playerMove('0',1, board);
+        main.playerMove('0',4, board);
+        main.playerMove('0',7, board);
+        assertTrue(main.evaluator(board), "review evaluator at positions 1,4,7");
+    }
+
+    /**
+     * evaluates if the game is over at win for Row line
+     */
```

Comment:
And what happens if the player completes row 4,5,6 or 7,8,9? will he/she win as well?

Reply:
added test to cover